wemake_python_styleguide/visitors/ast/imports.py

```diff
@@ -73,60 +62,120 @@ def check_alias(self, node: AnyImport) -> None:
                         VagueImportViolation(node, text=alias.name),
                     )
 
-    def check_same_alias(self, node: AnyImport) -> None:
+    def _check_same_alias(self, node: AnyImport) -> None:
         for alias in node.names:
             if alias.asname == alias.name and not self._options.i_control_code:
                 self._error_callback(
                     SameAliasImportViolation(node, text=alias.name),
                 )
 
-    def check_protected_import(self, node: AnyImport) -> None:
-        import_names = [alias.name for alias in node.names]
-        for name in chain(imports.get_import_parts(node), import_names):
+
+@final
+class _ImportValidator(_BaseImportValidator):
+    """Validator of ``ast.Import`` nodes."""
+
+    def validate(self, node: AnyImport) -> None:
+        node = cast(ast.Import, node)
```

Comment:
I dislike this casting :D
Do you know if there is any better way to type such strict subclasses which provides implementation only for one type of the type from `Union` defined in base class?